src/workerd/io/compatibility-date.c++

```diff
@@ -32,6 +39,26 @@ struct CompatDate {
     return !(*this < other);
   }
 
+#ifdef _WIN32
+  static kj::Maybe<CompatDate> parse(kj::StringPtr text) {
+    if (!text.startsWith("2")) return nullptr;
+    if (text.size() != 10 || text[4] != '-' || text[7] != '-') {
+      return nullptr;
+    }
+    for (char c: text) {
+      if ((c < '0' || '9' < c) && c != '-') return nullptr;
+    }
+    uint year, month, day;
+    // TODO(soon): use `kj::parse` here instead or some Windows time parsing API
+    auto result = sscanf(text.cStr(), "%d-%d-%d", &year, &month, &day);
```

Comment:
I wonder if perhaps this should have been the implementation on Linux too... Looking at it, I'm not really sure why I used a time parser when we really just want the numbers out. Thoughts?